src/Microsoft.ML.PCA/PcaTrainer.cs

```diff
@@ -469,6 +470,20 @@ public void SaveAsText(TextWriter writer, RoleMappedSchema schema)
             }
         }
 
+        public IDataView GetSummaryDataView(RoleMappedSchema schema)
+        {
+            var bldr = new ArrayDataViewBuilder(Host);
+
+            bldr.AddColumn("Mean vector", NumberType.R4, _mean);
+            bldr.AddColumn("Projected mean vector", NumberType.R4, _meanProjected);
+            for (var i = 0; i < _rank; ++i)
+            {
+                bldr.AddColumn("V" + i, NumberType.R4, _eigenVectors[i]);
```

Comment:
Indeed. Either this should be a single multidimensional vectror column (with `_rank` as the second dimension), or you should find a way to span this across multiple rows. (I prefer the first option myself, since honestly this ought to be part of metdata.)

---
In reply to: [189336699](https://github.com/dotnet/machinelearning/pull/185#discussion_r189336699) [](ancestors = 189336699)

Reply:
added as a single vector column